### PR TITLE
Add end-to-end test to verify unlisted packages in registration and search

### PR DESCRIPTION
**Pull request by joelverhagen**
Progress on https://github.com/NuGet/NuGetGallery/issues/3786.

### CODE REVIEW

File: src/NuGet.Services.EndToEnd/Support/Clients/GalleryClient.cs [hunk 1]
```diff
@@ -34,5 +34,20 @@ public GalleryClient(TestSettings testSettings)
...
             }
         }
+
+        public async Task UnlistAsync(string id, string version)
+        {
+            var url = $"{_testSettings.GalleryBaseUrl}/api/v2/package/{id}/{version}";
+            using (var httpClient = new HttpClient())
+            using (var request = new HttpRequestMessage(HttpMethod.Delete, url))
+            {
+                request.Headers.Add("X-NuGet-ApiKey", _testSettings.ApiKey);
```
Comment (skofman1) [inline src/NuGet.Services.EndToEnd/Support/Clients/GalleryClient.cs]: const for X-NuGet-ApiKey

Comment (joelverhagen) [inline src/NuGet.Services.EndToEnd/Support/Clients/GalleryClient.cs]: Good idea, thanks.

File: src/NuGet.Services.EndToEnd/Support/TestSettings.cs [hunk 2]
```diff
@@ -8,6 +8,20 @@ namespace NuGet.Services.EndToEnd.Support
 {
     public class TestSettings
     {
+        public enum Mode
+        {
+            EnvironmentVariables,
```
Comment (skofman1) [inline src/NuGet.Services.EndToEnd/Support/TestSettings.cs]: >EnvironmentVariables [](start = 12, length = 20)

consider renaming to "Local"

Comment (joelverhagen) [inline src/NuGet.Services.EndToEnd/Support/TestSettings.cs]: I thought of this but decided against because local is confusing since that could suggest localhost gallery or search.

File: src/NuGet.Services.EndToEnd/Support/Clients/GalleryClient.cs (hunk 1 above)
Comment (joelverhagen) [inline src/NuGet.Services.EndToEnd/Support/Clients/GalleryClient.cs]: Fixed.

**Comment by skofman1** [comment]
>             string apiKey,

Does it make sense to have API key as part of the test settings? You can't hard coded it bellow (and are forced to use an empty value), and there might be several API keys to be used in different scenarios.

---
Refers to: src/NuGet.Services.EndToEnd/Support/TestSettings.cs:29 in 03f417a. [](commit_id = 03f417a3da64efd8a18c8be49a9907893b5aadd7, deletion_comment = False)

**Comment by joelverhagen** [comment]
> Does it make sense to have API key as part of the test settings? You can't hard coded it bellow (and are forced to use an empty value), and there might be several API keys to be used in different scenarios.

Spoke offline about this. We decided to add multiple API keys to the framework when the need arises. In general, this sort of edge case testing should be done in functional test projects for specific components (as NuGetGallery already has).
